### DISCUSSION TITLE
How do I mock a derived atom in react-testing-library

### DISCUSSION THREAD

**Discussion by patricktruong-cb** [pmndrs/jotai#2327] [Q&A]
How can I mock a selector for instance is

`const getValues = atom((get) => get(ATOM)?.values);`

I have been trying to find a way to mock this for tests but using HydrateAtoms to setting the default value of ATOM and trying to mock the mockImplementation for useAtom but none work. Can I get some help

**Comment by dai-shi** [comment]
> using HydrateAtoms to setting the default value of ATOM

That sounds good to me.

> trying to mock the mockImplementation for useAtom

I've never heard someone did this before.

**Comment by dvd900** [comment]
I left a comment here with a person having a similar issue, not sure if people will like this but may help unblock you: https://github.com/pmndrs/jotai/discussions/1613#discussioncomment-8017697
